src/managers/XCursorManager.cpp

```diff
@@ -0,0 +1,448 @@
+    "box_spiral",
+    "center_ptr",
+    "circle",
+    "clock",
+    "coffee_mug",
+    "cross",
+    "cross_reverse",
+    "crosshair",
+    "diamond_cross",
+    "dot",
+    "dotbox",
+    "double_arrow",
+    "draft_large",
+    "draft_small",
+    "draped_box",
+    "exchange",
+    "fleur",
+    "gobbler",
+    "gumby",
+    "hand1",
+    "hand2",
+    "heart",
+    "icon",
+    "iron_cross",
+    "left_ptr",
+    "left_side",
+    "left_tee",
+    "leftbutton",
+    "ll_angle",
+    "lr_angle",
+    "man",
+    "middlebutton",
+    "mouse",
+    "pencil",
+    "pirate",
+    "plus",
+    "question_arrow",
+    "right_ptr",
+    "right_side",
+    "right_tee",
+    "rightbutton",
+    "rtl_logo",
+    "sailboat",
+    "sb_down_arrow",
+    "sb_h_double_arrow",
+    "sb_left_arrow",
+    "sb_right_arrow",
+    "sb_up_arrow",
+    "sb_v_double_arrow",
+    "shuttle",
+    "sizing",
+    "spider",
+    "spraycan",
+    "star",
+    "target",
+    "tcross",
+    "top_left_arrow",
+    "top_left_corner",
+    "top_right_corner",
+    "top_side",
+    "top_tee",
+    "trek",
+    "ul_angle",
+    "umbrella",
+    "ur_angle",
+    "watch",
+    "xterm",
+};
+// clang-format on
+
+std::vector<SP<SXCursors>> CXCursorManager::loadStandardCursors(std::string const& name, int size) {
+    std::vector<SP<SXCursors>> newCursors;
+
+    // load the default xcursor shapes that exist in the theme
+    for (size_t i = 0; i < XCURSOR_STANDARD_NAMES.size(); ++i) {
+        std::string shape{XCURSOR_STANDARD_NAMES.at(i)};
+        auto        xImages = XcursorShapeLoadImages(i << 1 /* wtf xcursor? */, name.c_str(), size);
+
+        if (!xImages) {
+            Debug::log(WARN, "XCursor failed to find a shape with name {}, trying size 24.", shape);
+            xImages = XcursorShapeLoadImages(i << 1 /* wtf xcursor? */, name.c_str(), 24);
+
+            if (!xImages) {
+                Debug::log(WARN, "XCursor failed to find a shape with name {}, skipping", shape);
+                continue;
+            }
+        }
+
+        auto cursor = createCursor(shape, xImages);
+        newCursors.emplace_back(cursor);
+
+        if (!defaultCursor && (shape == "left_ptr" || shape == "arrow"))
+            defaultCursor = cursor;
+
+        XcursorImagesDestroy(xImages);
+    }
+
+    // broken theme.. just set it.
+    if (!newCursors.empty() && !defaultCursor)
+        defaultCursor = newCursors.front();
+
+    return newCursors;
+}
+
+std::vector<SP<SXCursors>> CXCursorManager::loadAllFromDir(std::string const& path, int size) {
+    std::vector<SP<SXCursors>> newCursors;
+    std::string                full;
+
+    if (std::filesystem::exists(path) && std::filesystem::is_directory(path)) {
+        for (const auto& entry : std::filesystem::directory_iterator(path)) {
+            if (!entry.is_regular_file() && !entry.is_symlink())
+                continue;
+
+            full    = entry.path().string();
+            FILE* f = fopen(full.c_str(), "r");
```

Comment:
std::unique_ptr so you can add a dtor with `fclose` (see execandget)

Reply:
yeah changed to that.